rpc/utils.go

```diff
@@ -44,10 +44,16 @@ func isExported(name string) bool {
 }
 
 // Is this type exported or a builtin?
-func isExportedOrBuiltinType(t reflect.Type) bool {
+func isExportedOrBuiltinType(method string, t reflect.Type) bool {
 	for t.Kind() == reflect.Ptr {
 		t = t.Elem()
 	}
+	// big.Int types are disallowed, as we need to hexencode them properly. Using
+	// them anywhere is a programming error. We hard panic because it's impossible
+	// to reliably catch these type issues otherwise.
+	if t.Name() == "Int" && t.PkgPath() == "math/big" {
+		panic(fmt.Sprintf("big.Int disallowed in RPC (%s)", method))
+	}
```

Comment:
I don't like this because the RPC server shouldn't be concerned with such details. Imagine documenting this: "This RPC server can be used for any API, but your methods may not return *big.Int".

Reply:
This was the only way I myself could ensure that we caught all API violations and didn't break anything (i.e. I only fixed getBalance and this mechanism caught the other 2-3 occurrences).